ui/components/app/incoming-trasaction-toggle/hooks.js

```diff
@@ -0,0 +1,73 @@
+import { useState, useEffect } from 'react';
+import { TEST_CHAINS } from '../../../../shared/constants/network';
+
+export const useIncomingTransactionToggle = ({
```

Comment:
Originally i wanted to separate it to make the file smaller and more readable. easier to manage as well. If i change the name to `useIncomingTransactionToggleHook`, could it be a compromised solution?

Reply:
see [1ba6cb1f7d489bea929d9b6969e964406b94543e](https://github.com/MetaMask/metamask-extension/pull/20363/commits/1ba6cb1f7d489bea929d9b6969e964406b94543e)

Reply:
I can only speak from my opinions, but separating logic out into a separate file does not make it easier to read. We used to do this with our component/container pattern. You'd have to constantly switch between files to understand how things connect together. 

Though, a quick check of the hooks/ folder shows many that we are using only in one place. So here's what i'd ask:

Either
A) Inline the hook into the file it is used in. 
B) Move it to the hooks/ folder so that we at least preserve one pattern and we can then discuss how we want to handle this in the future.

Reply:
Solved by inline hooks from another pr https://github.com/MetaMask/metamask-extension/pull/20374